### ISSUE TITLE
Fullscreen freeze after 2.0.4 Update

### ISSUE THREAD

**Issue by LaloSoda** [OpenEmu/OpenEmu#2888]
2014 MacBook Pro - MacOS Sierra 10.12.1
After downloading the newest OpenEmu update (2.0.4), I am experiencing crashes after playing a game on fullscreen view then stopping the emulation and opening up another game. The sound seems to play, but the screen stays black and clicking on anything brings up the spinning wheel. 
If I keep the game on non-fullscreen view, I am able to switch between games without a problem.

**Comment by clobber** [comment]
Can't replicate that, and not sure there's any code touched between 2.0.3 and the 2.0.4 update that could cause it. We will need exact steps from you to replicate the problem, including the games you are using, in order to take a look.

**Comment by LaloSoda** [comment]
Here's a link to a quick video showing the problem:
https://youtu.be/OPMw-u6awIk
I tried all kinds of variations; going between games on different consoles, minimizing the view before stopping the emulation, starting and not starting from save points. If the view is fullscreened, the crash will happen when switching between games.

**Comment by clobber** [comment]
Have you tried restarting your machine and just launching OpenEmu to see if the problem persists? Some of us have followed your video and cannot replicate this problem on our end.

Beyond that, try changing the filter to Pixellate or Nearest Neighbor and make sure the games are using the Nestopia core plugin for NES games.

**Comment by LaloSoda** [comment]
I've restarted several times since the problem began, switched between both filters, Nestopia is checked off as the default core for NES games, and the problem still persists. 
Oh well, will have to quit the program between games for now one then. A small nuisance. I'm glad it seems to be only happening to me though.
Thanks for looking into this for me!

**Comment by robotoad** [comment]
If you go back to 2.0.3, does the problem persist?

**Comment by Floyde79** [comment]
I am having the same exact problem. I am running 2.0.4, Mac OS X El Capitan 10.11.6. It happens after every game. I have to force quit OpenEmu and start fresh. I have to do this for NES & SNES games as these are the only ones I have. It does not matter which game, it will do it every time.

I have noticed that this is only happening in when going back and forth from full screen. I don't mind a window that much, but it is a lot easier for the kids to follow the game when there are no outside distractions.

If you need any more info from me, just let me know. Or if there is anything you want me to check/test, let me know as well.

**Comment by robotoad** [comment]
@Floyde79 Can you [download 2.0.3](https://github.com/OpenEmu/OpenEmu/releases/download/v2.0.3/OpenEmu_2.0.3.zip) and confirm that the problem doesn't occur? We can't replicate the issue, and it will help to verify that it's specific to 2.0.4.

**Comment by PokeManiac4459** [comment]
I cant use my  controls on 2.0.4, I've restarted consecutively but It's doing nothing

**Comment by PokeManiac4459** [comment]
I cant change* ;P

**Comment by clobber** [comment]
@PokeManiac4459 Open a new issue. What you're talking about is completely unrelated to this.

**Comment by LaloSoda** [comment]
I just downloaded and ran 2.0.3, and switching between games after being on fullscreen goes smoothly. I switched between NES games and between NES and SNES and other games without a problem after being on fullscreen.

**Comment by Floyde79** [comment]
I did the same, with same results.

**Comment by Floyde79** [comment]
And now 2.0.4 is working also.

Is there a dependency of some type missing from 2.0.4 that could be causing this for those that didn't have 2.0.3? I'm not a programmer, but I do a LOT of testing of software at work (proprietary software that interfaces with proprietary hardware), and I have seen this happen before between releases....usually something really stupid and simple.

**Comment by LaloSoda** [comment]
Interesting. My 2.0.4 is still not working.

**Comment by Floyde79** [comment]
@LaloSoda What if you change your view to full screen from the game library. View>Enter Full Screen.

Does this have any effect?

**Comment by LaloSoda** [comment]
Still freezes after changing view to full screen from the game library and switching to another game

**Comment by Floyde79** [comment]
I stand corrected. My 2.0.4 is still doing the same thing. 2.0.3 rocking solid though.

**Comment by robotoad** [comment]
Can both of you give the model and specs of your Macs?

**Comment by LaloSoda** [comment]
MacBook Pro (Retina, 13-inch, Mid 2014)
2.6 GHz Intel Core i5
8 GB 1600 MHz DDR3
Intel Iris 1536 MB

**Comment by pboudoin** [comment]
same thing here but with a white screen.

**Comment by clobber** [comment]
@pboudoin "Me too" posts aren't helpful. Better to post more details like OS version and system specs.

**Comment by gsnow** [comment]
same issue. recently upgraded to 2.0.4 and this exact issue started happening- have not tried re-downloading previous version but it looks i might have to. 

macOS Sierra version 10.12
iMac (27 inch late 2009)
3.06 GHz Intel Core 2 Duo
14 gb 1067 MHZ DDR3
ATI radeon HD 4670 256 MB

**Comment by Kerry-Maxwell** [comment]
Also having the lockup after each game/ force quit problem with 2.0.4 (639-g606b287-Release)

Sierra 10.12.1 (16B2555)
iMac (Retina 5K, 27-inch, Late 2015)
4 GHz Intel Core i7
32 GB 1867 MHz DDR3
AMD Radeon R9 M390 2048 MB

**Comment by gsnow** [comment]
Went back to 2.0.3 and it's solid

**Comment by JAStanton** [comment]
Same:

```
OpenEmu v2.0.4
639-g606b287-Release
no usb controllers
```

```
  Hardware Overview:
  Model Name:	MacBook Pro
  Model Identifier:	MacBookPro12,1
  Processor Name:	Intel Core i5
  Processor Speed:	2.7 GHz
  Number of Processors:	1
  Total Number of Cores:	2
  L2 Cache (per Core):	256 KB
  L3 Cache:	3 MB
  Memory:	16 GB
  Boot ROM Version:	MBP121.0167.B17
  SMC Version (system):	2.28f7
```

I have 2 games installed for Super Nintendo (SNES)
```
The Legend of Zelda: A Link to the Past
Super Mario World
```

Full stack trace:
```
Exception Name: NSInvalidArgumentException
Description: -[OEGameViewController touchBar]: unrecognized selector sent to instance 0x6080001541c0
User Info: (null)

0   CoreFoundation                      0x00007fffa586852b __exceptionPreprocess + 171
1   libobjc.A.dylib                     0x00007fffb9f40cad objc_exception_throw + 48
2   CoreFoundation                      0x00007fffa58e9d34 -[NSObject(NSObject) doesNotRecognizeSelector:] + 132
3   CoreFoundation                      0x00007fffa57dabf5 ___forwarding___ + 1061
4   CoreFoundation                      0x00007fffa57da748 _CF_forwarding_prep_0 + 120
5   OpenEmu                             0x0000000107bd8b4e _TFE7OpenEmuCSo20OEGameViewControllerP33_239303EE196DE72FE9C49BC03ADD36B116toggleFullScreenfGSqP__T_ + 110
6   OpenEmu                             0x0000000107bd8e7a _TToFE7OpenEmuCSo20OEGameViewControllerP33_239303EE196DE72FE9C49BC03ADD36B123gameplayControlsTouchedfGSqP__T__merged + 90
7   libsystem_trace.dylib               0x00007fffbaa5003d _os_activity_initiate + 61
8   AppKit                              0x00007fffa3b884e7 -[NSApplication(NSResponder) sendAction:to:from:] + 456
9   AppKit                              0x00007fffa36d8245 -[NSControl sendAction:to:] + 86
10  AppKit                              0x00007fffa36d816d __26-[NSCell _sendActionFrom:]_block_invoke + 136
11  libsystem_trace.dylib               0x00007fffbaa5003d _os_activity_initiate + 61
12  AppKit                              0x00007fffa36d80c5 -[NSCell _sendActionFrom:] + 128
13  AppKit                              0x00007fffa371a92a -[NSButtonCell _sendActionFrom:] + 98
14  libsystem_trace.dylib               0x00007fffbaa5003d _os_activity_initiate + 61
15  AppKit                              0x00007fffa36d6a58 -[NSCell trackMouse:inRect:ofView:untilMouseUp:] + 2481
16  AppKit                              0x00007fffa371a667 -[NSButtonCell trackMouse:inRect:ofView:untilMouseUp:] + 785
17  AppKit                              0x00007fffa36d54c8 -[NSControl mouseDown:] + 832
18  AppKit                              0x00007fffa3ce873d -[NSWindow(NSEventRouting) _handleMouseDownEvent:isDelayedEvent:] + 6341
19  AppKit                              0x00007fffa3ce4f8c -[NSWindow(NSEventRouting) _reallySendEvent:isDelayedEvent:] + 1942
20  AppKit                              0x00007fffa3ce442a -[NSWindow(NSEventRouting) sendEvent:] + 541
21  AppKit                              0x00007fffa3b84bf5 -[NSApplication(NSEvent) sendEvent:] + 1145
22  OpenEmu                             0x0000000107c0398d _TTSf4g_n___TFC7OpenEmu18OpenEmuApplication9sendEventfCSo7NSEventT_ + 77
23  OpenEmu                             0x0000000107c03415 _TToFC7OpenEmu18OpenEmuApplication9sendEventfCSo7NSEventT_ + 37
24  AppKit                              0x00007fffa346b009 -[NSApplication run] + 1002
25  AppKit                              0x00007fffa3435a8a NSApplicationMain + 1237
26  OpenEmu                             0x0000000107bcde59 main + 73
27  libdyld.dylib                       0x00007fffba81e255 start + 1
```


Full repro steps:
- Launch OpenEMU
- Double click on game
- Click Full Screen

What happens?
<img width="591" alt="screen shot 2017-01-05 at 9 36 04 pm" src="https://cloud.githubusercontent.com/assets/645300/21708569/ff27b948-d38e-11e6-9713-fcb07b528856.png">

If I click continue the game keeps playing. Then when I take it out of fullscreen I get the same error prompt.

If I was to wager a guess as to what is happening. It is trying to update the touchBar on my mac, but I don't have one, so it explodes. Thats just a wild ass guess :)

Hope this helped!

**Comment by JAStanton** [comment]
@robotoad it looks like this is closed but when I check for updates I don't have any. Can you cut a new release with the fix?

**Comment by clobber** [comment]
@JAStanton The commit referenced right above your first post states it's been fixed and this issue was closed because of it. An update will be released including the fix when we're ready for 2.0.5.
